Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -1321,6 +1322,20 @@ protected void init(UUID choosingPlayerId) {
             addPlane(plane, startingPlayerId);
             state.setPlaneChase(this, gameOptions.planeChase);
         }
+
+        if (!gameOptions.perPlayerEmblemCards.isEmpty()) {
+            for (UUID playerId : state.getPlayerList(startingPlayerId)) {
+                for (Card card : gameOptions.perPlayerEmblemCards) {
+                    addEmblem(new EmblemOfCard(card), card, playerId);
```

Comment:
User must upload a simple data, not original cards due security reason. Also you must create card objects on the server by card name and number -- do not take original objects from a user uploaded data. See example from `createDeckDraftCube`, `CubeFromDeck` and `Deck->getDeckCardLists()`. Another possible usage style from a player's decks -- `DeckCardInfo`.

Reply:
Changed. Note that previously the user was uploading `MockCard`s, but I suppose a user with a modified client could upload non-mock cards... not sure exactly what that could have done.